### PR TITLE
MTN use reusable workflow and replace deprecated to_dict

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**

### CODE REVIEW

**Comment by mathurinm** [comment]
@Badr-MOUFAD can you investigate the test failure?

**Comment by Badr-MOUFAD** [comment]
@tomMoral could you trigger the CI?

**Comment by mathurinm** [comment]
@Badr-MOUFAD tests  seem to fail on your last push

File: benchmark_utils/utils.py [hunk 1]
```diff
@@ -1,17 +1,14 @@
-from benchopt import safe_import_context
-
-with safe_import_context() as import_ctx:
```
Comment (Badr-MOUFAD) [inline benchmark_utils/utils.py:3]: cf. https://github.com/benchopt/benchopt/issues/509, for why it was removed.

**Review by agramfort — APPROVED** [review]
@mathurinm merge if happy

**Comment by tomMoral** [comment]
I find it weird that there was an error, I would like to see what is the real traceback (NameError is simply due to the failed import).

I guess we can investigate later once we have more time as long as this does not cause a regression. Could you test that this does not make the autocompletion slow?

**Comment by mathurinm** [comment]
Just tested autocomplete, it looks OK

**Pull request merged**